In [66]:
import pandas as pd
import numpy as np


In [67]:
df=pd.read_csv('SOCR-HeightWeight.csv')# since we have this dataset with same location under jupiter notebook directly reading.
df

Index  Height(Inches)  Weight(Pounds)
0          1        65.78331        112.9925
1          2        71.51521        136.4873
2          3        69.39874        153.0269
3          4        68.21660        142.3354
4          5        67.78781        144.2971
...      ...             ...             ...
24995  24996        69.50215        118.0312
24996  24997        64.54826        120.1932
24997  24998        64.69855        118.2655
24998  24999        67.52918        132.2682
24999  25000        68.87761        124.8742

[25000 rows x 3 columns]

In [70]:
# Sort the dataframe by both Height and Weight in increasing order
sorted_df1 = df.sort_values(by=['Height(Inches)', 'Weight(Pounds)'], ascending=[True, True])
sorted_df1

Index  Height(Inches)  Weight(Pounds)
13971  13972        60.27836       110.11380
2651    2652        60.61265        88.04646
20608  20609        60.80620       113.91450
5641    5642        60.86340       106.19390
12031  12032        60.86977       108.86330
...      ...             ...             ...
15209  15210        74.59993       147.03720
13681  13682        74.74047       155.54620
16752  16753        74.84890       122.16640
2481    2482        75.11519       153.95620
1893    1894        75.15280       146.97010

[25000 rows x 3 columns]

In [68]:
df.shape

(25000, 3)

In [4]:
df.describe()

Index  Height(Inches)  Weight(Pounds)
count  25000.000000    25000.000000    25000.000000
mean   12500.500000       67.993114      127.079421
std     7217.022701        1.901679       11.660898
min        1.000000       60.278360       78.014760
25%     6250.750000       66.704397      119.308675
50%    12500.500000       67.995700      127.157750
75%    18750.250000       69.272958      134.892850
max    25000.000000       75.152800      170.924000

In [29]:
D1 = df.drop('Weight(Pounds)', axis='columns')
D1

Index  Height(Inches)
0          1        65.78331
1          2        71.51521
2          3        69.39874
3          4        68.21660
4          5        67.78781
...      ...             ...
24995  24996        69.50215
24996  24997        64.54826
24997  24998        64.69855
24998  24999        67.52918
24999  25000        68.87761

[25000 rows x 2 columns]

In [69]:
# Sort the dataframe by Height in increasing order
sorted_df = D1.sort_values(by='Height(Inches)', ascending=True)
sorted_df

Index  Height(Inches)  Weight(Pounds)
13971  13972        60.27836       110.11380
2651    2652        60.61265        88.04646
20608  20609        60.80620       113.91450
5641    5642        60.86340       106.19390
12031  12032        60.86977       108.86330
...      ...             ...             ...
15209  15210        74.59993       147.03720
13681  13682        74.74047       155.54620
16752  16753        74.84890       122.16640
2481    2482        75.11519       153.95620
1893    1894        75.15280       146.97010

[167 rows x 3 columns]

# we need to outlier for each column i'e for height and weight using the following mehod.

# for height
# IQR=Q3-Q1

In [31]:
Q1=df['Height(Inches)'].quantile(0.25)
Q3=df['Height(Inches)'].quantile(0.75)
Q1,Q3

(66.7043975, 69.2729575)

In [32]:
IQR=Q3-Q1
IQR

2.568560000000005

# now we will going to find lower_limit and upper_limit by using the following methods

In [33]:
lower_limit=Q1-1.5*IQR  # lower_limit formula
upper_limit=Q3+1.5*IQR  # upper_limit formula
lower_limit,upper_limit

(62.85155749999999, 73.1257975)

# The value for (Height(Inches)) which are less than lower_limit((62.85155749999999) and greater than upper_limit(73.1257975) will be cosidered as outliers.

# using the following methods we can get outliers

In [65]:
D1_outlier = df[(df['Height(Inches)'] < lower_limit) | (df['Height(Inches)'] > upper_limit)]
D1_outlier


Index  Height(Inches)  Weight(Pounds)
0          1        65.78331        112.9925
1          2        71.51521        136.4873
2          3        69.39874        153.0269
3          4        68.21660        142.3354
4          5        67.78781        144.2971
...      ...             ...             ...
24995  24996        69.50215        118.0312
24996  24997        64.54826        120.1932
24997  24998        64.69855        118.2655
24998  24999        67.52918        132.2682
24999  25000        68.87761        124.8742

[25000 rows x 3 columns]

# the above values are outliers which will remove among dataset for height

In [49]:
D1_no_outlier= df[(df['Height(Inches)'] >lower_limit) & (df['Height(Inches)'] < upper_limit)]
D1_no_outlier

Index  Height(Inches)  Weight(Pounds)
0          1        65.78331        112.9925
1          2        71.51521        136.4873
2          3        69.39874        153.0269
3          4        68.21660        142.3354
4          5        67.78781        144.2971
...      ...             ...             ...
24995  24996        69.50215        118.0312
24996  24997        64.54826        120.1932
24997  24998        64.69855        118.2655
24998  24999        67.52918        132.2682
24999  25000        68.87761        124.8742

[24833 rows x 3 columns]

# second step removing outliers for weight parameters among dataset

In [51]:
Q1=df['Weight(Pounds)'].quantile(0.25)
Q3=df['Weight(Pounds)'].quantile(0.75)
Q1,Q3

(119.308675, 134.89285)

In [52]:
# evaluating inter quartile range IQR for weight parameters.
IQR=Q3-Q1
IQR

15.584175000000016

In [53]:
lower_limit=Q1-1.5*IQR  # lower_limit formula
upper_limit=Q3+1.5*IQR  # upper_limit formula
lower_limit,upper_limit

(95.93241249999997, 158.26911250000003)

# removing outliers which has value less than lower_limit(95.93241249999997) and upper_limit(158.26911250000003))

In [64]:
D2_outlier = df[(df['Weight(Pounds)'] < lower_limit) | (df['Weight(Pounds)'] > upper_limit)]
D2_outlier

Index  Height(Inches)  Weight(Pounds)
156      157        70.98112       158.95620
249      250        65.93895        84.35980
293      294        69.13438        92.74955
367      368        72.32489       168.22900
446      447        69.16807       162.41090
...      ...             ...             ...
24187  24188        69.21155       161.85040
24610  24611        66.56804        82.38298
24750  24751        70.82192       159.63440
24835  24836        65.17339        94.40946
24889  24890        71.01048       161.85320

[191 rows x 3 columns]

# the above values are outliers which have extreme value and will be removed from the original data.

In [56]:
# getting dataset for weight which has no outliers
D2_no_outlier= df[(df['Weight(Pounds)'] >lower_limit) & (df['Weight(Pounds)'] < upper_limit)]
D2_no_outlier


Index  Height(Inches)  Weight(Pounds)
0          1        65.78331        112.9925
1          2        71.51521        136.4873
2          3        69.39874        153.0269
3          4        68.21660        142.3354
4          5        67.78781        144.2971
...      ...             ...             ...
24995  24996        69.50215        118.0312
24996  24997        64.54826        120.1932
24997  24998        64.69855        118.2655
24998  24999        67.52918        132.2682
24999  25000        68.87761        124.8742

[24809 rows x 3 columns]

# check another method read the vedio

In [ ]:
import pandas as pd
import numpy as np

# Assuming df is your dataframe
# Calculate Z-scores for height and weight columns
df['height_zscore'] = (df['height'] - df['height'].mean()) / df['height'].std()
df['weight_zscore'] = (df['weight'] - df['weight'].mean()) / df['weight'].std()

# Set threshold for outliers
threshold = 3

# Identify outliers
outliers = df[(np.abs(df['height_zscore']) > threshold) | (np.abs(df['weight_zscore']) > threshold)]

# Remove outliers
df_no_outliers = df.drop(outliers.index)


In [ ]:
# the abve code is used to remove outliers from the dataset and remains with 